test/test.js

```diff
@@ -166,35 +653,295 @@ suite('GaiaHeader', function() {
+    });
+
+    test('It uses `window.innerWidth` by default', function(done) {
+      this.dom.innerHTML = `<gaia-header action="back">
+        <h1>Title</h1>
+      </gaia-header>`;
+
+      var el = this.dom.firstElementChild;
+
+      afterNext(el, 'runFontFit').then(() => {
+        sinon.assert.called(this.innerWidth_spy);
+        sinon.assert.notCalled(this.clientWidth_spy.get, 'queried DOM');
+      }).then(done, done);
+    });
+
+    test('It uses `gaiaHeader.clientWidth` when the `not-flush` attribute is used', function(done) {
+      this.dom.innerHTML = `<gaia-header action="back" not-flush>
+        <h1>Title</h1>
+      </gaia-header>`;
+
+      var el = this.dom.firstElementChild;
+
+      afterNext(el, 'runFontFit').then(() => {
+        sinon.assert.called(this.clientWidth_spy.get, 'queried DOM');
+      }).then(done, done);
+    });
+  });
+
+  test('It works with multiple headers', function(done) {
+    this.dom.innerHTML = `<gaia-header action="back">
+      <h1 class="title-1">Title 1</h1>
+      <h1 class="title-2" hidden>Title 2</h1>
+      <button></button>
+      <button></button>
+    </gaia-header>`;
+
+    var el = this.dom.firstElementChild;
+    var title1 = el.querySelector('.title-1');
+    var title2 = el.querySelector('.title-2');
+
+    afterNext(el, 'runFontFit').then(() => {
+      assert.equal(title1.style.MozMarginStart, '50px');
+      assert.equal(title2.style.MozMarginStart, '50px');
+    }).then(() => {
+      this.fontFit.reset();
+      title1.hidden = true;
+      title2.hidden = false;
+      return afterNext(el, 'runFontFit').then(() => {
+        sinon.assert.notCalled(this.fontFit, 'font-fit not required');
+        assert.equal(title1.style.MozMarginStart, '50px');
+        assert.equal(title2.style.MozMarginStart, '50px');
+      });
+    }).then(done, done);
+  });
+
+  test('it clears pending async tasks when the element is detached', function(done) {
+    this.dom.innerHTML = '<gaia-header action="menu"><h1>Title</h1></gaia-header>';
+    var el = this.dom.firstElementChild;
+
+    el.remove();
+
+    afterNext(el, 'runFontFit').then(() => {
+      done(new Error('should not have run'));
+    }).catch(done, done);
+  });
+
+  test('it clears pending async `setTitleStyles`', function(done) {
+    this.dom.innerHTML = '<gaia-header action="menu"><h1>Title</h1></gaia-header>';
+    var el = this.dom.firstElementChild;
+
+    el.runFontFit();
+    el.remove();
+
+    afterNext(el, 'setTitleStyle').then(() => {
+      done(new Error('should not have run'));
+    }).catch(done, done);
+  });
+
+  /**
+   * Utils
+   */
+
+  function afterNext(obj, method) {
+    var wait = 100;
+    var timeout;
+
+    return new Promise((resolve, reject) => {
+      var real = obj[method];
+
+      // If the function doesn't run
+      // after `wait` period: reject.
+      timeout = setTimeout(() => {
+        obj[method] = real;
+        reject();
```

Comment:
reject with an error, otherwise you'll call "done" without an error in the tests, and mocha won't be able to determine that it's an error